KubeArmor/feeder/policyMatcher.go

```diff
@@ -12,440 +13,338 @@ import (
 // == Security Policies == //
 // ======================= //
 
-// UpdateSecurityPolicies Function
-func (fd *Feeder) UpdateSecurityPolicies(action string, conGroup tp.ContainerGroup) {
-	if action == "DELETED" {
-		delete(fd.SecurityPolicies, conGroup.NamespaceName+"_"+conGroup.ContainerGroupName)
-	} else { // ADDED | MODIFIED
-		matches := tp.MatchPolicies{}
-
-		for _, secPolicy := range conGroup.SecurityPolicies {
-			if len(secPolicy.Spec.Process.MatchPaths) > 0 {
-				for _, path := range secPolicy.Spec.Process.MatchPaths {
-					if len(path.FromSource) == 0 {
-						match := tp.MatchPolicy{}
-
-						match.PolicyName = secPolicy.Metadata["policyName"]
-
-						match.Severity = strconv.Itoa(path.Severity)
-						match.Tags = path.Tags
-						match.Message = path.Message
-
-						match.Source = ""
-						match.Operation = "Process"
-						match.Resource = path.Path
-
-						match.Action = path.Action
-
-						matches.Policies = append(matches.Policies, match)
-					} else {
-						for _, src := range path.FromSource {
-							if len(src.Path) > 0 {
-								match := tp.MatchPolicy{}
-
-								match.PolicyName = secPolicy.Metadata["policyName"]
-
-								match.Severity = strconv.Itoa(path.Severity)
-								match.Tags = path.Tags
-								match.Message = path.Message
-
-								match.Source = src.Path
-								match.Operation = "Process"
-								match.Resource = path.Path
-
-								match.Action = path.Action
-
-								matches.Policies = append(matches.Policies, match)
-							} else if len(src.Directory) > 0 {
-								match := tp.MatchPolicy{}
-
-								match.PolicyName = secPolicy.Metadata["policyName"]
+func newProcPathMatchPolicy(policyName, source string, path tp.ProcessPathType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(path.Severity),
+		Tags:       path.Tags,
+		Message:    path.Message,
+		Source:     source,
+		Operation:  "Process",
+		Resource:   path.Path,
+		Action:     path.Action,
+	}
```

Comment:
These are just helper functions. It could be managed through interfaces instead of helpers, but as a first proposal I preferred not to touch types.go concerning that. Another additional discussion would be related to the use of embedded structure/interface types. I noticed that some have similar structure.